From 1a76126a93ecc47358cd639f3911cd6ef7ca9128 Mon Sep 17 00:00:00 2001
From: Ruikai Liu <rickliu2000@outlook.com>
Date: Sat, 10 Jan 2026 16:07:55 +0800
Subject: [PATCH] KDECompilerSettings: disable --fatal-warnings in linker flags

We now have GCS warnings spamming the terminal when linking to shared
objects that don't come with GCS enabled. GCC 15 now default to enable
GCS when branch protection is either set to 'standard' or 'gcs'.
---
 kde-modules/KDECompilerSettings.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kde-modules/KDECompilerSettings.cmake b/kde-modules/KDECompilerSettings.cmake
index 98ce0433..d82fdc7f 100644
--- a/kde-modules/KDECompilerSettings.cmake
+++ b/kde-modules/KDECompilerSettings.cmake
@@ -558,10 +558,10 @@ endfunction()
 if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT APPLE) OR
         (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT APPLE) OR
         (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND NOT WIN32))
-    # Linker warnings should be treated as errors
-    set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--fatal-warnings ${CMAKE_SHARED_LINKER_FLAGS}")
-    set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--fatal-warnings ${CMAKE_MODULE_LINKER_FLAGS}")
 
+    # We now have GCS warnings scattering everywhere when linking with shared
+    # libraries that doesn't come with GCS enabled.
+    # Removing --fatal-warnings for the linker.
     string(TOUPPER "CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}" compileflags)
     if("${CMAKE_CXX_FLAGS} ${${compileflags}}" MATCHES "-fsanitize")
         set(sanitizers_enabled TRUE)
-- 
2.52.0

